File: Bit Manipulation/Bitwise-AND-of-Range-of-Numbers.py
class Solution:
  def solve(self, start, end):
    bitwise_and = current = start
    while current <= end:
      for bit in range(30):
        if current & (1 << bit):
          current += 1 << bit
          break
      else:
        break

      if current <= end:
        bitwise_and &= current

    return bitwise_and

File: Bit Manipulation/Bitwise-AND-of-Range-of-Numbers.cpp
int solve(int start, int end) {
  int bitwise_and = start, current = start;
  while (current <= end) {
    current += 1 << __builtin_ctz(current);
    if (current <= end) {
      bitwise_and &= current;
    }
  }
  return bitwise_and;
}